logstash-core/lib/logstash/instrument/metric_store.rb

```diff
@@ -0,0 +1,94 @@
+# encoding: utf-8
+require "concurrent"
+require "logstash/event"
+
+module LogStash module Instrument
+  # The Metric store the data structure that make sure the data is
+  # saved in a retrievable way, this is a wrapper around multiples ConcurrentHashMap
+  # acting as a tree like structure.
+  class MetricStore
+    class ConcurrentMapExpectedError < Exception; end
```

Comment:
It's unlikely to happen but still can happen.

I should never happen to plugins developers since their metrics instance will be correctly configured and scope for their plugin at runtime. its mostly act as a guards.